File: solutions/153.find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const findMin = nums => {
  let min = nums[0]
  for (let i = 1; i < nums.length; i++) {
    let item = nums[i]
    if (item < min) return item

    min = Math.min(min, item)
  }

  return min
}

File: solutions/153.find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        [4,5,6,7,0,1,2]
         l   m       r

        1 2 3 4 5 0
            l m   r
                lm r


        [3,4,5,1,2]
               l r

        3 1 2
        l m r
        """
        l = 0
        r = len(nums) - 1

        while l < r:
            m = l + (r - l) // 2

            if nums[m] > nums[r]:
                l = m + 1
            else:
                r = m

        return nums[r]
